common/txmgr/confirmer.go

```diff
@@ -584,7 +592,8 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) fet
 			return fmt.Errorf("saveFetchedReceipts failed: %w", err)
 		}
 		// Save the receipts but mark the associated transactions as Fatal Error since the original transaction was purged
-		if err := ec.txStore.SaveFetchedReceipts(ctx, purgeReceipts, TxFatalError, ec.stuckTxDetector.StuckTxFatalError(), ec.chainID); err != nil {
+		stuckTxFatalErrMsg := ec.stuckTxDetector.StuckTxFatalError()
```

Comment:
nit:
would it be better to have `TerminallyStuckMsg` defined as var instead of const, so you don't have to change the `StuckTxFatalError()` signature, and don't need to line 595 and 632

Reply:
I think using `TerminallyStuckMsg` directly would introduce an evm import in the common code which is why we have to serve this message through the stuck tx detector

Reply:
Oh I was talking about the `errors.go`, instead of `const TerminallyStuckMsg = "transaction terminally stuck"` can we have `var TerminallyStuckMsg = "transaction terminally stuck"`.

This way you can just take addr of the var in `stuck_tx_detector.go` instead of refactoring func signature

Reply:
Oh I see, we could do it that way too. I was also updating the signature because I was thinking returning `*string` was also a little misleading. It makes it seem this error message could be `nil` but it never would be.